File: frontend/src/types/firstRun.ts
export type FirstRunStepStatus = 'pending' | 'passed' | 'warning' | 'failed'
export type ReadinessProfileName = 'development_ready' | 'demo_ready' | 'field_dry_run_ready' | 'hardware_telemetry_ready' | 'competition_rehearsal_ready' | 'release_candidate_ready'
export type ReadinessProfileStatus = 'not_evaluated' | 'passed' | 'warning' | 'failed' | 'blocked'
export type CurrentFirstRunStatus = 'open' | 'passed' | 'warning' | 'failed'

export interface FirstRunStep {
  step_id: string
  title: string
  status: FirstRunStepStatus
  explanation: string
  suggested_fix: string | null
  blocking: boolean
  detail: Record<string, unknown>
}

export interface FirstRunReport {
  run_id: string
  created_at: number
  mode: string
  completed: boolean
  overall_status: FirstRunStepStatus
  steps: FirstRunStep[]
  summary: Record<string, unknown>
  profile_statuses: Record<ReadinessProfileName, ReadinessProfileStatus>
  profile_checklists: Record<ReadinessProfileName, FirstRunStep[]>
  report_path: string | null
  no_physical_command_generated: boolean
}

export interface FirstRunStatus {
  completed: boolean
  latest_report: FirstRunReport | null
  mode: string
  checks_count: number
  current_first_run_status: CurrentFirstRunStatus
  current_profile_id: ReadinessProfileName
  current_profile_evaluation_status: ReadinessProfileStatus
  last_successful_first_run: {
    run_id: string
    profile_id: ReadinessProfileName
    timestamp: number
    checks_count: number
  } | null
  stale_evidence: boolean
  no_physical_command_generated: boolean
}

export interface FirstRunActionResult {
  accepted: boolean
  reason: string
  status: FirstRunStatus
}

File: backend/app/schemas/first_run.py
import time
from typing import Any, Literal

from pydantic import BaseModel, Field


FirstRunStepStatus = Literal["pending", "passed", "warning", "failed"]
ReadinessProfileName = Literal[
    "development_ready",
    "demo_ready",
    "field_dry_run_ready",
    "hardware_telemetry_ready",
    "competition_rehearsal_ready",
    "release_candidate_ready",
]
ReadinessProfileStatus = Literal["not_evaluated", "passed", "warning", "failed", "blocked"]
CurrentFirstRunStatus = Literal["open", "passed", "warning", "failed"]


class FirstRunStep(BaseModel):
    step_id: str
    title: str
    status: FirstRunStepStatus
    explanation: str
    suggested_fix: str | None = None
    blocking: bool = False
    detail: dict[str, Any] = Field(default_factory=dict)


class FirstRunReport(BaseModel):
    run_id: str
    created_at: float = Field(default_factory=time.time)
    mode: str
    completed: bool = False
    overall_status: FirstRunStepStatus = "pending"
    steps: list[FirstRunStep] = Field(default_factory=list)
    summary: dict[str, Any] = Field(default_factory=dict)
    profile_statuses: dict[ReadinessProfileName, ReadinessProfileStatus] = Field(default_factory=dict)
    profile_checklists: dict[ReadinessProfileName, list[FirstRunStep]] = Field(default_factory=dict)
    report_path: str | None = None
    no_physical_command_generated: bool = True


class FirstRunStatus(BaseModel):
    completed: bool = False
    latest_report: FirstRunReport | None = None
    mode: str
    checks_count: int = 0
    current_first_run_status: CurrentFirstRunStatus = "open"
    current_profile_id: ReadinessProfileName = "release_candidate_ready"
    current_profile_evaluation_status: ReadinessProfileStatus = "not_evaluated"
    last_successful_first_run: dict[str, Any] | None = None
    stale_evidence: bool = False
    no_physical_command_generated: bool = True


class FirstRunActionResult(BaseModel):
    accepted: bool
    reason: str
    status: FirstRunStatus
